kernel/src/schema.rs

```diff
@@ -619,4 +632,33 @@ mod tests {
         let schema: Result<Schema, _> = serde_json::from_reader(file);
         assert!(schema.is_ok())
     }
+
+    #[test]
+    fn test_invalid_decimal() {
+        let data = r#"
+        {
+            "name": "a",
+            "type": "decimal(39, 10)",
+            "nullable": false,
+            "metadata": {}
+        }
+        "#;
+        assert!(matches!(
+            serde_json::from_str::<StructField>(data).unwrap_err(),
+            _
+        ));
+
+        let data = r#"
+        {
+            "name": "a",
+            "type": "decimal(10, 39)",
+            "nullable": false,
+            "metadata": {}
+        }
+        "#;
+        assert!(matches!(
+            serde_json::from_str::<StructField>(data).unwrap_err(),
+            _
+        ));
```

Comment:
I'm not grokking this syntax? What is the `_`?

Reply:
fixed.